### ISSUE TITLE
gRPC-python shouln't require service names to be enumerated when configuring reflection

### ISSUE THREAD

**Issue by dimo414** [grpc/grpc#34933]
C++ enables reflection application-wide by simply linking `libgrpc++_reflection` into the binary. Java and Go require instantiating a reflection service (`ProtoReflectionService` and `reflection`, respectively) but this similarly applies for all services installed in the server. By contrast Python requires passing a list of service names to `enable_server_reflection()`, which means the callsite must enumerate all services the server will host, which is often difficult or at least tedious to discern, and does not necessarily have to reflect reality (e.g. the strings could be wrong or out of sync with what is actually running on the service).

Given that gRPC clearly knows all services that are installed (in order to route incoming traffic in the first place) it seems needlessly cumbersome for Python server maintainers to have to keep a list of service names in-sync just to support reflection. Is it possible the Python reflection API could be improved to work like these other languages and not require hand-enumeration?

### Is your feature request related to a problem? Please describe.
It is particularly difficult to determine the set of services that will be enabled in a loosely-coupled framework architecture where individual components can register themselves with a server during initialization. Needing to track a centralized list of service names breaks that decomposition.

### Describe the solution you'd like
Drop the `service_name` argument from `enable_server_reflection()` and support whichever services are registered with the server :slightly_smiling_face: 

### Describe alternatives you've considered
I poked around for an API to query the server for its known services, but did not see anything obviously useful.

### Additional context
Referenced https://github.com/grpc/grpc/blob/master/doc/python/server_reflection.md as well as the other language documentation.

**Comment by gnossen** [comment]
@dimo414 You're completely right. I've had this thought since the moment I saw the Python reflection API. I don't think the change would be too difficult. The only real consideration is backward-compatibility. We need to continue supporting the current version that takes two arguments: an `Iterable[str]` and a `grpc.Server`. We could overload the first argument to be `Union[Iterable[str], grpc.Server]`, but that makes the API docs look a bit messy.

We could use the Linux model and call the new API `enable_server_reflection2` or maybe just `enable_reflection`? Regardless, someone is going to need to write up a gRFC about the backward-compatibility concerns and then do the typing. As always, PRs are appreciated.

**Comment by dimo414** [comment]
Can you share some code pointers for what would need to be changed, and in particular where the service names could be retrieved from? I haven't mucked with the gRPC codebase before but might be able to justify taking a crack at this.

Presumably adding a new function would be straightforward and backwards-compatible, as you say. Aside from picking a new name are there other particular compatibility issues you're concerned about?
